### ISSUE TITLE
Validate rule config

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#222]
## Summary

Currently, secretlint has basic validation using JSON Schema
https://github.com/secretlint/secretlint/tree/master/packages/%40secretlint/config-validator

However, following invalid config is passed.

```json
{
  "rules": [
    {
      "id": "@secretlint/secretlint-rule-preset-recommend",
      "rules": [
        {
          "id": "invalid-id"
        }
      ]
    }
  ]
}
```

The `invalid-id` does not exists in `@secretlint/secretlint-rule-preset-recommend`, but secretlint does not throw error.
It is just ignored.

### Basic example

secretlint should throw errors in thie case.

```json
{
  "rules": [
    {
      "id": "@secretlint/secretlint-rule-preset-recommend",
      "rules": [
        {
          "id": "invalid-id"
        }
      ]
    }
  ]
}
```

Supported case:

secretlint already throw an error in following case.

```json
{
  "rules": [
    {
       "id": "invalid-id"
    }
  ]
}
```

### Motivation

Make User friendly configuration.

### Related

https://github.com/secretlint/secretlint/issues/171
